src/main/plugins/websearch-plugin/websearch-plugin.ts

```diff
@@ -36,20 +38,46 @@ export class WebSearchPlugin implements ExecutionPlugin {
                         return -1;
                     }
                     return 0;
-                })
-                .map((webSearchEngine): SearchResultItem => {
-                    return {
-                        description: this.buildDescription(webSearchEngine, userInput),
-                        executionArgument: this.buildExecutionArgument(webSearchEngine, userInput),
+                });
+            const promises = webSearchEnginesArray.map((webSearchEngine) => {
+                return axios({ method: "GET", url: webSearchEngine.suggestionUrl.replace("{{query}}", this.getSearchTerm(webSearchEngine, userInput)) });
+            });
+
+            axios.all(promises).then(axios.spread((...responses) => {
+                let responseCounter = 0;
+                responses.forEach((res) => {
+                    const results = res.data[1];
+                    searchResults.push({
+                        description: this.buildDescription(webSearchEnginesArray[responseCounter], userInput),
+                        executionArgument: this.buildExecutionArgument(webSearchEnginesArray[responseCounter], userInput),
                         hideMainWindowAfterExecution: true,
-                        icon: isValidIcon(webSearchEngine.icon) ? webSearchEngine.icon : defaultWebSearchIcon,
-                        name: webSearchEngine.name,
+                        icon: isValidIcon(webSearchEnginesArray[responseCounter].icon) ? webSearchEnginesArray[responseCounter].icon : defaultWebSearchIcon,
+                        name: userInput.replace(webSearchEnginesArray[responseCounter].prefix, ""),
                         originPluginType: this.pluginType,
                         searchable: [],
-                    };
-                });
+                    });
```

Comment:
You don't need a variable `responseCounter` here, you get the current counter from the second parameter of the `forEach` function:

`responses.forEach(response, index) => ...`

Reply:
Additionally `res` is a bad variable name, I suggest you rename it to `response` so it is clear what it is.

Reply:
I will change it.